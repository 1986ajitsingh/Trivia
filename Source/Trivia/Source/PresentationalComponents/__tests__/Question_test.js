import renderer from 'react-test-renderer';
import React from 'react';
import ModuleUnderTest from '../Question';

describe('renders correctly', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange/ Act
    const mockFun = jest.fn();
    const tree = renderer.create(<ModuleUnderTest onQuestionAnswered={mockFun} />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});

describe('constructor', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange
    const expectedState = {
      selectedSegment: 'none',
    };

    // Act
    const moduleUnderTest = new ModuleUnderTest({});

    // Assert
    expect(moduleUnderTest.state).toEqual(expectedState);
  });
});

describe('componentWillReceiveProps', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange
    const moduleUnderTest = new ModuleUnderTest({});
    moduleUnderTest.state = 'some state';

    // Act
    moduleUnderTest.componentWillReceiveProps();

    // Assert
    expect(moduleUnderTest.state).toEqual({ selectedSegment: 'none' });
  });
});

describe('optionSelected', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange
    const mockSetState = jest.fn();
    const mockSetTimeout = jest.fn();
    const mockClearTimeout = jest.fn();
    const mockOnQuestionAnswered = jest.fn();
    const moduleUnderTest = new ModuleUnderTest({});
    const selectedSegment = 'some selected segment';
    moduleUnderTest.props = { onQuestionAnswered: mockOnQuestionAnswered };
    moduleUnderTest.setState = jest.fn((state, callback) => {
      mockSetState(state, callback);
      if (callback) { callback(); }
    });
    const originalSetTimeout = setTimeout;
    /* eslint-disable no-global-assign */
    setTimeout = jest.fn((callback, milliseconds) => {
      mockSetTimeout(callback, milliseconds);
      if (callback) { callback(); }
    });
    const originalClearTimeout = clearTimeout;
    clearTimeout = mockClearTimeout;

    // Act
    moduleUnderTest.optionSelected(selectedSegment);
    clearTimeout = originalClearTimeout;
    setTimeout = originalSetTimeout;
    /* eslint-enable no-global-assign */

    // Assert
    expect(mockSetState).toBeCalledWith({ selectedSegment }, expect.anything());
    expect(mockSetTimeout).toBeCalledWith(expect.anything(), 100);
    expect(mockOnQuestionAnswered).toBeCalledWith(selectedSegment);
    expect(mockClearTimeout).toBeCalled();
  });
});
