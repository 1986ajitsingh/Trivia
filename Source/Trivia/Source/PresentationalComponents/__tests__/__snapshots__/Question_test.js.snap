// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E6E6FA",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4B0082",
        "fontSize": 30,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    No question text given
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#4B0082",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "opacity": 1,
              "paddingBottom": 5,
              "paddingTop": 5,
            },
            Object {
              "flex": 1,
            },
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
              "marginBottom": 10,
              "marginTop": 10,
            },
            Object {
              "color": "#4B0082",
            },
          ]
        }
      >
        True
      </Text>
    </View>
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Array [
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "opacity": 1,
                "paddingBottom": 5,
                "paddingTop": 5,
              },
              Object {
                "flex": 1,
              },
            ],
            Object {
              "borderLeftColor": "#4B0082",
              "borderLeftWidth": 1,
            },
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "fontSize": 24,
              "marginBottom": 10,
              "marginTop": 10,
            },
            Object {
              "color": "#4B0082",
            },
          ]
        }
      >
        False
      </Text>
    </View>
  </View>
</View>
`;
